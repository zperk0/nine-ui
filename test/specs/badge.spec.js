import NBadge from '../../packages/n-badge';
import { mount } from '@vue/test-utils';

describe('NBadge', () => {
  it('renders correctly', () => {
    const wrapper = mount({
      render () {
        return <NBadge>Follow</NBadge>;
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('render default style when not define type', () => {
    const wrapper = mount(NBadge);
    expect(wrapper.classes()).toContain('n-badge');
  });
  it('render chinese characters correctly', () => {
    const wrapper = mount({
      render () {
        return <NBadge>测试</NBadge>;
      }
    });
    expect(wrapper.text()).toBe('测试');
  });
  it('render correctly content', () => {
    const wrapper = mount({
      render () {
        return <NBadge content="99" />;
      }
    });
    expect(wrapper.text()).toBe('99');
  });
});
