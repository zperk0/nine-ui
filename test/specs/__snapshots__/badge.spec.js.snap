// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NBadge renders correctly 1`] = `<div class="n-badge">Follow<span class="n-badge-common n-badge-default"></span></div>`;
